x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationIndexService.java

```diff
@@ -101,6 +102,7 @@ public class SearchApplicationIndexService {
     public final NamedWriteableRegistry namedWriteableRegistry;
     private final BigArrays bigArrays;
 
+    @Inject
     public SearchApplicationIndexService(
         Client client,
         ClusterService clusterService,
```

Comment:
@afoucret hello! kathleen suggested i reach out to you and ask if you know of a better way of making this `SearchApplicationIndexService` injectable. we looked at adding an instance in the return value of [`createComponents`](https://github.com/sloanelybutsurely/elasticsearch/blob/search-applications-telemetry/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearch.java#L150) in `EnterpriseSearch.java` but noticed we didn't have a `BigArrays` instance available.

do you know how we could get a `BigArrays` instance or perhaps some better way?

Reply:
Potentially `BigArrays.NON_RECYCLING_INSTANCE`? But I'm not familiar enough with this part of our code to confidently say that's a good suggestion.

Reply:
I remember @jimczi advocating for this service to be kept as non-injectable as it is a thin abstraction that wasn't worth to extract as a service (we started having that as a service and it was changed). I'm struggling to find that conversation though 😓 

 Is there any change to the usage of the service to revert that decision back? (disclaimer: I would be a huge fan of doing it 😅 )

Reply:
If we're using the service to provide telemetry information that may be enough of a justification - however, if we figure out the correct way to add BigArrays we can create it in `EnterpriseSearch` and return it so it's accessible where it needs to be. I don't have super strong feelings about it other than the hesitation to make the service injectable without hearing @jimczi 's feedback.

Reply:
We don't really need to call the service directly? Can we rely on the action instead, something like:
```
 client.execute(ListSearchApplicationAction.INSTANCE, ...
```